.github/workflows/build_and_test.yml

```
@@ -83,12 +83,17 @@ jobs:
         cd OpenDDS/ACE_TAO
         export ACE_COMMIT=$(git rev-parse HEAD)
         echo "ACE_COMMIT=$ACE_COMMIT" >> $GITHUB_ENV
+    - name: get Compiler version
+      shell: bash
+      run: |
+        export COMPILER_VERSION=$("c++ --version 2>&1 | head -n 1")
```

I don't think this will evaluate correctly with the quotes, and it will grab the whole first line of the version which can have spaces and possibly fill up the cache key more than we need (I believe there's a 256 byte character limit). Maybe something like:
```suggestion
        export COMPILER_VERSION=$(c++ --version |& head -n 1 | md5sum | cut -d ' ' -f 1)
```